xgi/convert.py

```diff
@@ -207,14 +225,30 @@ def convert_to_simplicial_complex(data, create_using=None):
             (ee.members(e), e, deepcopy(attr)) for e, attr in ee.items()
         )
         H._hypergraph = deepcopy(data._hypergraph)
+        return H
+    
+    elif isinstance(data, Hypergraph):
+        H = empty_simplicial_complex(create_using)
+        H.add_nodes_from((n, attr) for n, attr in data.nodes.items())
+        ee = data.edges
+        H.add_simplices_from(
+            (ee.members(e), e, deepcopy(attr)) for e, attr in ee.items()
+        )
+        return H
 
     elif isinstance(data, list):
         # edge list
-        from_hyperedge_list(data, create_using)
+        if not isinstance(create_using, SimplicialComplex):
+            return convert_to_simplicial_complex(from_hyperedge_list(data, create_using))
+        else:
+            from_hyperedge_list(data, create_using)
 
     elif isinstance(data, pd.DataFrame):
-        from_bipartite_pandas_dataframe(data, create_using)
-
+        if not isinstance(create_using, SimplicialComplex):
+            return convert_to_simplicial_complex(from_bipartite_pandas_dataframe(data, create_using))
+        else:
+            from_bipartite_pandas_dataframe(data, create_using)
+            
```

Comment:
Prefer fewer calls:
```suggestion
        result = from_bipartite_pandas_dataframe(data, create_using)
        if not isinstance(create_using, SimplicialComplex):
            return convert_to_simplicial_complex(result)            
```

Reply:
Thank you for the feedbacks @leotrs, I've just pushed the commit with all your corrections included